SEQLinkage/Core.py

```diff
@@ -666,8 +687,52 @@ def run(self):
                 break
 
 # Cell
+def get_family_with_var(data):
+    items = []
+    for item,item_vars in data.famvaridx.items():
+        if len(item_vars) == 0:  #no variants in the family
+            for person in data.families[item]:
+                data[person] = ("0", "0")
+        else:
+            items.append(item)
+    return items
+
+haplotyper = cstatgen.HaplotypingEngine(verbose = env.debug)
+def phasing_haps(chrom,item,fvar,fgeno):
+    item_varnames, positions, item_mafs = fvar
+    try:
+        item_haplotypes = haplotyper.Execute(chrom, item_varnames, positions, fgeno)[0]
+    except:
+        print(item,"fail to phase haplotypes")
+        item_haplotypes = []
+    item_haplotypes = np.array(item_haplotypes)
+    return item,item_varnames,item_mafs,item_haplotypes
+
+def test(region,data,extractor,maker,writer):
```

Comment:
this is a test function and should not be in the code. If you want to have test functions can you make those separate python scripts? eg something like test.py that you can tell from the name that it is not part of the core features?

Reply:
removed